File: oep_spring_backend/src/main/java/com/oep/service/UserService.java
package com.oep.service;

import java.util.List;

import com.oep.dtos.*;
import com.oep.entities.User;

import jakarta.validation.Valid;

public interface UserService {

	void createUser(CreateUserDto dto);

	List<UserResponseDto> getAllUsers();
	
	List<com.oep.dtos.UserResponseDto> getUsersByRole(String role);

	UserResponseDto getUserById(Long id);

	void updateUser(Long id, UpdateUserDto dto);

	void deleteUser(Long id);

}

File: frontend-Application/src/services/admin/userService.js
import adminAxios from "../axios/adminAxios";

export const getAllUsers = async () => {
    return await adminAxios.get("/admin/users");
};

export const getUserById = async (id) => {
    return await adminAxios.get(`/admin/users/${id}`);
};

export const getUsersByRole = async (role) => {
    return await adminAxios.get(`/admin/users?role=${role}`);
};

export const getAllInstructors = async () => {
    return await getUsersByRole("instructor");
};

export const updateUser = async (id, userData) => {
    return await adminAxios.put(`/admin/users/${id}`, userData);
};

export const deleteUser = async (id) => {
    return await adminAxios.delete(`/admin/users/${id}`);
};
